pkg/tool/osdctl/osdctl.go

```diff
@@ -0,0 +1,241 @@
+	source *github.Source
+}
+
+func NewTool() *Tool {
+	t := &Tool{
+		source: github.NewSource("openshift", "osdctl"),
+	}
+	return t
+}
+
+func (t *Tool) Name() string {
+	return "osdctl"
+}
+
+func (t *Tool) Install(rootDir, latestDir string) error {
+	// Pull latest release from GH
+	release, err := t.source.FetchLatestRelease()
+	if err != nil {
+		return err
+	}
+
+	// Determine which assets to download
+	var checksumAsset gogithub.ReleaseAsset
+	var osdctlBinaryAsset gogithub.ReleaseAsset
+	var arch = runtime.GOARCH
+	if arch == "amd64" {
+		arch = "x86_64"
+	}
+	for _, asset := range release.Assets {
+		if asset.GetName() == "sha256sum.txt" {
+			if checksumAsset.GetName() != "" {
+				return fmt.Errorf("detected duplicate osdctl checksum assets")
+			}
+			checksumAsset = *asset
+			continue
+		}
+		if !strings.Contains(asset.GetName(), arch) {
+			continue
+		}
+		if !strings.Contains(strings.ToLower(asset.GetName()), strings.ToLower(runtime.GOOS)) {
+			continue
+		}
+
+		if osdctlBinaryAsset.GetName() != "" {
+			return fmt.Errorf("detected duplicate osdctl binary asset")
+		}
+		osdctlBinaryAsset = *asset
+	}
+	// Ensure both checksum and binary were retrieved
+	if checksumAsset.GetName() == "" || osdctlBinaryAsset.GetName() == "" {
+		return fmt.Errorf("failed to find osdctl or it's checksum")
+	}
+
+	// Download the arch- & os-specific assets
+	toolDir := t.toolDir(rootDir)
+	versionedDir := filepath.Join(toolDir, release.GetTagName())
+	err = os.MkdirAll(versionedDir, os.FileMode(0755))
+	if err != nil {
+		return fmt.Errorf("failed to create version-specific directory '%s': %w", versionedDir, err)
+	}
+
+	err = t.source.DownloadReleaseAssets([]*gogithub.ReleaseAsset{&checksumAsset, &osdctlBinaryAsset}, versionedDir)
+	if err != nil {
+		return fmt.Errorf("failed to download one or more assets: %w", err)
+	}
+
+	// Verify checksum of downloaded assets
+	osdctlBinaryFilepath := filepath.Join(versionedDir, osdctlBinaryAsset.GetName())
+	fileBytes, err := os.ReadFile(osdctlBinaryFilepath)
+	if err != nil {
+		return fmt.Errorf("failed to read osdctl binary file '%s' while generating sha256sum: %w", osdctlBinaryFilepath, err)
+	}
+	sumBytes := sha256.Sum256(fileBytes)
+	// TODO - there's probably a better way to do this
+	binarySum := fmt.Sprintf("%x", sumBytes[:])
+
+	checksumFilePath := filepath.Join(versionedDir, checksumAsset.GetName())
+	checksumFile, err := os.Open(checksumFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to open osdctl checksum file '%s': %w", checksumFilePath, err)
+	}
+	var checksum string
+	scanner := bufio.NewScanner(checksumFile)
+	for scanner.Scan() {
+		line := scanner.Text()
+		fmt.Printf("'%s'\n", line)
+		tokens := strings.Fields(line)
+		if len(tokens) != 2 {
+			return fmt.Errorf("the checksum file '%s' is invalid", checksumFile.Name())
+		}
+
+		if osdctlBinaryAsset.GetName() != tokens[1] {
+			continue
+		}
+		checksum = tokens[0]
+	}
+	if err = checksumFile.Close(); err != nil {
```

Comment:
Lets split this out into two lines to remain consistent with the rest of the code